### DISCUSSION TITLE
How do we contribute to the docs by updating snippets?

### DISCUSSION THREAD

**Discussion by ryaino** [excaliburjs/Excalibur#2698] [Q&A] [answered]
After going through the "getting started guide", I found a small issue with one of the snippets. I tried having a look at the docs repo myself to update but I'm assuming the snippets are set up using sub modules or something? I've never actually used that feature before so I'm pretty confused about how I can create a PR for an updated snippet. 

Thanks for any help.

**Comment by eonarheim** [comment, marked as answer]
Hi @ryaino! Good question!

Some of the snippets are linked in from a separate repo into the doc via a special comment that the docs repo looks for in the submodule. Others are just inlined to the mdx, slightly arbitrary unfortunately. But the end goal is to know when the Excalibur API breaks something in the docs automatically because the submodules will start failing builds from the latest excalibur.

The special comment in the submodules looks like this `// start-snippet{some-label}` with code in between `// end-snippet{some-label}`

Here is an example of the docs using that label to embed the snippet from the submodule https://github.com/excaliburjs/excaliburjs.github.io/blob/site/docs/02-getting-started.mdx#L84

To make a change to one of these code samples open a PR in either of the current repos, the change will automatically be pulled into the docs repo when it's merged
* snippets https://github.com/excaliburjs/excalibur-snippets 
* breakout https://github.com/excaliburjs/sample-breakout
